packages/create-cloudflare/src/helpers/poll.ts

```diff
@@ -54,20 +57,43 @@ export const poll = async (url: string): Promise<boolean> => {
 	return false;
 };
 
-async function dnsLookup(domain: string): Promise<boolean> {
+export const isDomainAvailable = async (domain: string) => {
 	try {
-		const resolver = new Resolver({ timeout: TIMEOUT, tries: 1 });
-		resolver.setServers([
-			"1.1.1.1",
-			"1.0.0.1",
-			"2606:4700:4700::1111",
-			"2606:4700:4700::1001",
-		]);
-		return (await resolver.resolve4(domain)).length > 0;
-	} catch (e) {
+		const nameServers = await lookupAuthoritativeServers(domain);
+		if (nameServers.length === 0) return false;
+
+		const dns = new dns2({ nameServers });
+		const res = await dns.resolve(domain, "A");
+		return res.answers.length > 0;
+	} catch (error) {
 		return false;
 	}
-}
+};
+
+// Looks up the nameservers that are responsible for this particular domain
+export const lookupAuthoritativeServers = async (domain: string) => {
+	const nameServers = await lookupRootNameservers(domain);
+	const dns = new dns2({ nameServers });
+	const res = (await dns.resolve(domain, "NS")) as DnsResponse;
+
+	return (
+		res.authorities
+			// Filter out non-authoritative authorities (ones that don't have an 'ns' property)
+			.filter((r) => Boolean(r.ns))
+			// Return only the hostnames of the authoritative servers
+			.map((r) => r.ns)
+	);
+};
+
+// Looks up the nameservers responsible for handling `pages.dev` domains
+export const lookupRootNameservers = async (domain: string) => {
```

Comment:
I don't know what is the right terminology for these nameservers, but it's as far as I know it's not root, as root it only the very top name server (https://www.cloudflare.com/learning/dns/glossary/dns-root-server/).

Maybe call this `lookupParentNameServers` to avoid future confusion?